doc/source/development/dispatching.rst

```diff
@@ -0,0 +1,93 @@
+Dispatching
+===========
+
+The dispatching mechanism allows users to use alternative implementations of the algorithms
+available in scikit-image. The alternative implementations are provided by separate
+Python packages maintained outside of scikit-image. Example use-cases are support for array
+libraries other than Numpy or optimised implementations for particular hardware.
+
+This section of the documentation describes how to create an alternative implementation (a "backend")
+and how the dispatching mechanism in scikit-image works.
+
+.. important::
+    Dispatching support and infrastructure is experimental and is not ready for production.
+    It's made available as an early draft so that developers can gain experience
+    with the system.
+
+    Expect the dispatching API to change without notice.
+
+
+Implementing a new backend
+--------------------------
+
+An alternative implementation ("backend") is a good place to provide optimized implementations
+for particular hardware, support array libraries other than Numpy or explore novel ideas that
+are not (yet) a good fit for the core scikit-image library.
+
+To create a backend you have to create a new Python package that registers two particular
+`entrypoints <https://packaging.python.org/en/latest/specifications/pyproject-toml/#entry-points>`_. The first entrypoint called `skimage_backends` should resolve to a namespace
+that contains two functions: `can_has(name, *args, **kwargs)` and `get_implementation(name)`.
+The second entrypoint called `skimage_backend_infos` should resolve to a function with no
+arguments that returns an instance of the `skimage.util._backends.BackendInformation` class
+or something that behaves like it.
+
+
+The `skimage_backend_infos` entrypoint
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The information your backend provides via this entrypoint is used to get things like its
+name, homepage and which functions it implements. The main requirement for this entrypoint
+is that it is fast to import and has no dependencies other than `scikit-image`.
+
+The return value of the function should be an instance of the
+`skimage.util._backends.BackendInformation` class or something that behaves like it.
```

Comment:
I'd like to keep it to make it so that you can read this section and get all the information about the `skimage_backend_infos` entry point. Even at the price of repetition.